README.md

```diff
@@ -677,20 +677,101 @@ In either case:
     `{...}`, but they should ask themselves if this code is really readable and
     whether the block's content can be extracted into nifty methods.
 
-* Avoid `return` where not required.
+* Highly recommend to use `return` even if it is not required in Ruby language.
+
+    Although it is a Ruby way that an explicit return is unnecessary, it introduces
+    confusions to programmers working with unfamilar code and to programmers learning
+    Ruby from other strict languages.
 
     ```Ruby
-    # bad
+    # bad: it is unclear whether this function want to return the updated value
+    #      or just want to set the new value
+    def update(new_value)
+      @value = func(new_value]
+    end
+
+    # Someone may extend the function by add a log message, and then unintentionally
+    # complete break it if the return value is intended
+    def update(new_value)
+      @value = func(new_value]
+      Rails.logger.info '...'
+    end
+
+    # good: make it clear the return value is intended
+    def update(new_value)
+      @value = func(new_value]
+      return @value
```

Comment:
You could also take out the parentheses. But I think it's best the way @garygangwu wrote it.